engine/gui/widgets.py

```diff
@@ -89,10 +91,83 @@ def create(
             cancel_action=cancel_action,
             initial_selected_button=selected_button,
         )
+
+        cls.validate(gui_spec)
+        return gui_spec
+
+    def validate(self) -> None:
+        """Validates the GUISpec."""
+
+        if not self.buttons and not self.images:
+            raise ValidationError("Must have at least one button or image")
+
+        asset_names = collections.Counter([a.name for a in self.assets])
+        duplicate_assets = [asset for asset, count in asset_names.items() if count > 1]
+        if duplicate_assets:
+            raise ValidationError(f"Duplicate asset name: {duplicate_assets}.")
+
+        button_names = collections.Counter([b.name for b in self.buttons])
+        duplicate_buttons = [name for name, count in button_names.items() if count > 1]
+        if duplicate_buttons:
+            raise ValidationError(f"Duplicate button name: {duplicate_buttons}")
+
+        image_names = collections.Counter([i.image_asset for i in self.images])
+        duplicate_images = [image for image, count in image_names.items() if count > 1]
+        if duplicate_images:
+            raise ValidationError(f"Duplicate image name(s): {duplicate_images}")
+
+        for b_names in button_names.items():
+            if b_names[0] not in list(asset_names.elements()):
+                raise ValidationError(f"Button {b_names[0]} not in assets")
+
+        for i_names in image_names.items():
+            if i_names[0] not in list(asset_names.elements()):
+                raise ValidationError(f"Image {i_names[0]} not in assets")
+
+        # for name in asset_names.elements():
+        #     if name not in [b for b in button_names.elements()]:
+        #         raise ValidationError(f"Asset {name} not in buttons")
+        #     if name not in [i for i in self.images]:
+        #         raise ValidationError(f"Asset {name} not in images")
+
+        for a in self.assets:
+            if not pathlib.Path(a.path).is_file():
+                raise ValidationError(f"Asset path does not exist: {a.path}")
+
+        for button in self.buttons:
+            if button.center[0] < 0 or button.center[0] > core.SCREEN_WIDTH:
+                raise ValidationError(
+                    f"Button {button.name} is off screen horizontally"
+                )
+            if button.center[1] < 0 or button.center[1] > core.SCREEN_HEIGHT:
+                raise ValidationError(f"Button {button.name} is off screen vertically")
+            for direction in ["left", "right", "up", "down"]:
+                name = getattr(button, direction)
+                if name and name not in button_names:
+                    raise ValidationError(
+                        f"Button {name} referenced by {direction} not in buttons"
+                    )
+
+        if self.initial_selected_button not in self.buttons:
+            raise ValidationError("Initial selected button not in buttons")
```

Comment:
Consider raising this earlier in the code. In fact, you can get it to be raised automatically just by removing the `None` argument to `next()` in `GUISpec.create()`:

```python
selected_button = next(b for b in buttons if b.name == selected_button_name)
```
This will result in the system raising an exception if the button doesn't exist, which is what we want.

Even better is if we translate the exception to `ValidationError`:

```python
try:
  selected_button = next(b for b in buttons if b.name == selected_button_name)
except StopIteration:
  raise ValidationError(f"Initial selected button {selected_button_name} doesn't exist")
```